Formula/xorg-server.rb

```diff
@@ -0,0 +1,135 @@
+  depends_on "libxkbfile"  => :build
+  depends_on "meson"       => :build
+  depends_on "ninja"       => :build
+  depends_on "pkg-config"  => :build
+  depends_on "util-macros" => :build
+  depends_on "xorgproto"   => :build
+  depends_on "xtrans"      => :build
+
+  depends_on "libxfixes"
+  depends_on "libxfont2"
+  depends_on "mesa"
+  depends_on "pixman"
+  depends_on "xauth"
+  depends_on "xcb-util"
+  depends_on "xcb-util-image"
+  depends_on "xcb-util-keysyms"
+  depends_on "xcb-util-renderutil"
+  depends_on "xcb-util-wm"
+  depends_on "xkbcomp"
+  depends_on "xkeyboardconfig"
+
+  on_macos do
+    depends_on "gnu-getopt"
+    depends_on "libapplewm"
+  end
+
+  on_linux do
+    depends_on "dbus"
+    depends_on "libdrm"
+    depends_on "libepoxy"
+    depends_on "libxcvt"
+    depends_on "libxshmfence"
+    depends_on "nettle"
+    depends_on "systemd"
+  end
+
+  resource "xvfb-run" do
+    url "https://salsa.debian.org/xorg-team/xserver/xorg-server/-/raw/xorg-server-2_21.1.4-1/debian/local/xvfb-run"
+    sha256 "fd05e0f8e6207c3984b980a0f037381c9c4a6f22a6dd94fdcfa995318db2a0a4"
+  end
+
+  resource "xvfb-run.1" do
+    url "https://salsa.debian.org/xorg-team/xserver/xorg-server/-/raw/xorg-server-2_21.1.4-1/debian/local/xvfb-run.1"
+    sha256 "08f14f55e14e52e5d98713c4d8f25ae68d67e2ee188dc0247770c6ada6e27c05"
+  end
+
+  def install
+    # ChangeLog contains some non relocatable strings
+    rm "ChangeLog"
+    meson_args = %W[
+      --prefix=#{HOMEBREW_PREFIX}
+      --buildtype=release
+      --wrap-mode=nofallback
+
+      -Dxephyr=true
+      -Dxf86bigfont=true
+      -Dxcsecurity=true
+
+      -Dbundle-id-prefix=#{Formula["xinit"].plist_name.chomp ".startx"}
+      -Dbuilder_addr=#{tap.remote}
+      -Dbuilder_string=#{tap.name}
+    ]
+    # macOS doesn't provide `authdes_cred` so `secure-rpc=false`
+    # glamor needs GLX with `libepoxy` on macOS
+    if OS.mac?
+      meson_args += %w[
+        -Dsecure-rpc=false
+        -Dapple-applications-dir=libexec
+      ]
+    end
+
+    # X11.app need startx etc. in the same directory
+    destdir = buildpath/"dest"
+    system "meson", "build", *meson_args
+    system "meson", "compile", "-C", "build"
+    system "meson", "install", "-C", "build", "--destdir", destdir
+    prefix.install Dir["#{destdir}#{HOMEBREW_PREFIX}/*"]
+    # follow https://github.com/XQuartz/XQuartz/blob/main/compile.sh#L955
+    bin.install_symlink bin/"Xquartz" => "X" if OS.mac?
+
+    bin.install resource("xvfb-run")
+    man1.install resource("xvfb-run.1")
+    if OS.mac?
+      inreplace bin/"xvfb-run" do |s|
+        s.gsub! "getopt", Formula["gnu-getopt"].opt_bin/"getopt"
+        s.gsub!(/(?<=fmt) -t/, "") # bsd fmt doesn't support -t
+        s.change_make_var! "MCOOKIE", "$(openssl rand -hex 16)"
+      end
+    end
+  end
+
+  def caveats
+    <<~EOS
+      To launch X server, it is recommend to install xinit,
+      especially on macOS, otherwise X11.app will not work:
+        brew install xinit
```

Comment:
Should we install it by default? 
Are we installing a `.app` here? If so: perhaps we should not?

Reply:
I think we should consider removing the `.app` bundle here. We generally try to avoid them with formula.

Reply:
Thanks for letting me know. I guess I'm wondering what makes this XQuartz preferable to the cask?

Reply:
Can you use X11 on macOS without the XQuartz-related formulae here and just e.g. running `xinit`?

Reply:
Ok, sorry: is there another non-`.app`/XQuartz path to running a working X server (perhaps with `brew services`) on macOS?

Reply:
xvfb feels very non-native on macOS - I tested that when this PR was at an early stage. Quartz uses Apple's UI frameworks to provide native windowing.

Reply:
@Bo98 Ok, thanks, I understand now. Is it possible to run `xquartz` through `brew services` without the `.app`?

Reply:
Using Apple's UI frameworks quite often means bundling in an `.app` format, especially if nibs (Xcode Interface Builder) is involved. So no unless you want to modify the code to try do something special.

The code that builds the app actually lives in xorg-server itself rather than xquartz-base or anything like that.

Reply:
Ok, fine with me in this case then 👍🏻